### ISSUE TITLE
[Action Button] Slow Response Time on Android

### ISSUE THREAD

**Issue by iamkyleee** [xotahal/react-native-material-ui#112]
![ezgif com-optimize](https://cloud.githubusercontent.com/assets/7822506/23587582/a6f02e5a-0164-11e7-842c-c061f89e5b0e.gif)

I hope this GIF helped explain the delay. If not, it takes at least 2 seconds when clicking each button before it reacts.

I'm usin RN 0.42.0 (but it was the same with 0.41) and RN MUI 1.8.0

**Comment by mdeprezzo** [comment]
For me is the same. On back button on toolbar, when i press the back button. Possible bug ?

**Comment by xotahal** [comment]
Hey guys, I don't have this problem with the same components. I think you guys try to do something very complex within onPress callback, don't you? If you remove onPress callback, will it work properly?

**Comment by NewOldMax** [comment]
Did you try to turn off dev mode? It works fast for me with RN 0.42 and RN MUI 1.8.0

**Comment by xotahal** [comment]
@mdeprezzo What navigator do you use? Is the scene, where you navigate, empty? Do you fetch/compute something within onComponentDidMount/onComponentWillMount/etc methods?

It's really hard to say where the problem is. But everything works for me. And everything works fine also in [demo app](https://github.com/xotahal/react-native-material-ui-demo-app).

**Comment by mdeprezzo** [comment]
@xotahal I'm using react-native-router-flux, but like said before @NewOldMax, i turn off dev mode, and now, on virtual device, seems to works normally.

But yes, when i click to come back to the first scene, i invoke an action from redux to fetch data.
In demo app dev mode is on ?

**Comment by xotahal** [comment]
In demo app it works quite fast in dev mode as well. Think there is a problem with navigate and fetching data in same time. Look at InteractionManager in React Native's doc ;)

**Comment by mdeprezzo** [comment]
I don't think that, there is something wrong. Before using this package, i've used ToolbarAndroid from RN. And all works fine. But anyway, i'll check again the work flow.

**Comment by DeanHristov** [comment]
Hi, everyone 
I have the same problem in the Android emulator (Genymotion). At the moment I use the latest version of RN and I cannot decide what exactly is the problem. On the real device, the action button works a bit faster. 

The code below is a piece of code: 

```
    <ActionButton
      active={state.isActive}
      onPress={() => setState({isActive: !state.isActive})}
      buttonColor={Colors.orange}>
      {actions.map((item, idx: number) => (
        <ActionButton.Item
          key={`action-item-${idx}`}
          size={item.size ? item.size : 42}
          buttonColor={item.color ? item.color : Colors.success}
          title={item.title} onPress={item.onPress}>
          <Icon name={item.icon} color={item.iconColor ? item.iconColor : Colors.white} size={item.size ? item.size : 18} />
        </ActionButton.Item>
      ))}
    </ActionButton>
```   
Best, 
D. Hristov
